### PR TITLE
Add "Trip completed" State

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds a trip completed status so that when someone arrives at a destination, appropriate interactions can be made available.

### CODE REVIEW

**Review by JymDyerIBI — APPROVED** [review]
👍  Looks good.

File: src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java [hunk 1]
```diff
@@ -178,111 +193,159 @@ void canNotRestartAnOngoingJourney() throws Exception {
...
+                "ARRIVED: Monroe Dr NE at Cooledge Ave NE",
+                TripStatus.COMPLETED,
+                "Instructions for destination coordinate"
+            ),
+            Arguments.of(
+                multiLegMonitoredTrip,
+                createPoint(multiItinSecondLegDestCoords, 1, NORTH_WEST_BEARING),
+                "Get off here (Piedmont Ave NE at Monroe Dr)",
+                TripStatus.BEHIND_SCHEDULE,
+                "FIXME: No instruction for bus stop (code assumes we are on the transit leg already)."
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java]: Can this be fixed as part of this update? Is it in scope?

**Review by br648 — COMMENTED** [review]
This is fine. Just one comment on the "FIXME". If that is out of scope, let me know and I'll approve.

File: src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java]: @br648  It is not in scope, but I think I mixed up the leg walking to a bus stop and the leg riding the bus. I'll fix that.

Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java]: Addressed in https://github.com/ibi-group/otp-middleware/pull/257/commits/47df409b4cf60612bc28395e483a29570f64d8a8.

**Review by br648 — APPROVED** [review]
Thanks for the update. Looks good!

**Review by JymDyerIBI — COMMENTED** [review]
👍  New changes look fine.

**Pull request merged**